app/models/filters/hud_filter_base.rb

```diff
@@ -30,6 +30,14 @@ def effective_project_ids
           pluck(p_t[:id])
       end
 
+      if coc_codes.present?
+        @effective_project_ids = GrdaWarehouse::Hud::ProjectCoc.in_cocs(coc_codes: coc_codes).
+          where(ProjectID: @effective_project_ids).
+          distinct.
+          pluck(:ProjectID).
```

Comment:
ProjectID isn't correct, this needs to be `pluck(p_t[:id])`.  ProjectID is a string and is the internal foreign key for the particular data source.

Reply:
ah this again. okay. is this right? spot checked with some specific projects
```
      @effective_project_ids = GrdaWarehouse::Hud::ProjectCoc.in_coc(coc_code: coc_codes).
          joins(:project).
          where(p_t[:id].in(@effective_project_ids)).
          distinct.
          pluck(p_t[:id])
```

Reply:
That looks good.  I'd missed `where(ProjectID: @effective_project_ids).`  Thanks!